### ISSUE TITLE
Updates from side tags get stuck if newer builds are in the base tag

### ISSUE THREAD

**Issue by mohanboddu** [fedora-infra/bodhi#3991]
Updates like
https://bodhi.fedoraproject.org/updates/FEDORA-2020-641e69f08c and
https://bodhi.fedoraproject.org/updates/FEDORA-2020-d120de33c5
are stuck since there is a build in the side tag that is older (build time not nvr) compared to what is in the base tag.

Before side tags, when releng used to create the side tags, when we merge the side tags back into the base tag, we always considered the latest build, so if the side tag has a older build we would skip tagging it.

Maybe we should consider that option and comment in the update saying that we skipped tagged the older builds and let maintainers handle how they want to handle after that.

**Comment by cverna** [comment]
This is the way we intended bodhi to managed the side tag, if when trying to merge we find a more recent build in the target tag, we push the update back to pending and there is a comment on the update flagging which builds are conflicting.

The packager then needs to edit the update to either add a more recent build in the side tag or remove the old build. 

I am not sure we want Bodhi to be too smart in what it is trying to do, I think leaving the packager dealing with the conflict is better.

Thoughts ?

**Comment by hroncok** [comment]
I suppose updating the build list will reset the timer and karma, correct? Than before it can reach stable, another packages are outdated. With large updates, this can pretty much never end.

**Comment by cverna** [comment]
> I suppose updating the build list will reset the timer and karma, correct? Than before it can reach stable, another packages are outdated. With large updates, this can pretty much never end.

Yeah sorry I should have mentioned that this is for rawhide, so karma and days to stable are both set to 0 in rawhide.

**Comment by mattiaverga** [comment]
> This is the way we intended bodhi to managed the side tag, if when trying to merge we find a more recent build in the target tag, we push the update back to pending and there is a comment on the update flagging which builds are conflicting.
> 
> The packager then needs to edit the update to either add a more recent build in the side tag or remove the old build.
> 
> I am not sure we want Bodhi to be too smart in what it is trying to do, I think leaving the packager dealing with the conflict is better.
> 
> Thoughts ?

Considering that other updates in the sidetag can depend / have been built against the older build, if we just let Bodhi ignore the older build and push the others we may end up in having broken packages. So, yes, I think it's better to warn the packager and let him fix the conflicting builds.

**Comment by hroncok** [comment]
> Yeah sorry I should have mentioned that this is for rawhide, so karma and days to stable are both set to 0 in rawhide.

Even there, with very large updates, it will take some time to add/remove the build and in the meantime, people will rebuild. The problem will not be that bad, but I still see it. However, maybe we can "solve" it with documentation.
